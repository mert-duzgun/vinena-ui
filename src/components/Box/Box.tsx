import * as React from 'react';
import { PolymorphicComponentPropWithRef, PolymorphicRef } from 'types';

type BoxProps<C extends React.ElementType> = PolymorphicComponentPropWithRef<C, {}>;

type BoxComponent = <C extends React.ElementType = 'div'>(
  props: BoxProps<C>
) => React.ReactElement | null;

export const Box: BoxComponent = React.forwardRef(function VinenaBox<
  C extends React.ElementType = 'div'
>({ as: component, children, ...rest }: BoxProps<C>, ref?: PolymorphicRef<C>) {
  const Element = component || 'div';

  return (
    <Element ref={ref} {...rest}>
      {children}
    </Element>
  );
});
