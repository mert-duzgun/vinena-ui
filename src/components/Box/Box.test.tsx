import { render } from '@testing-library/react';

import { Box } from './Box';

describe('Vinena/components/Box', () => {
  it('should match snapshot', async () => {
    const { container } = render(<Box>Test</Box>);
    await expect(container.firstChild).toMatchSnapshot();
  });

  it('renders a div by default', () => {
    const { container } = render(<Box>Test</Box>);
    expect(container.firstChild).toBeInstanceOf(HTMLDivElement);
  });

  it('renders the component with custom props', () => {
    const { container } = render(
      <Box as="span" className="test">
        Test
      </Box>
    );
    expect(container.firstChild).toHaveClass('test');
    expect(container.firstChild).toBeInstanceOf(HTMLSpanElement);
  });
});
