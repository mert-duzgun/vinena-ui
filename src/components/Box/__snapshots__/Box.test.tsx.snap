// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Vinena/components/Box > should match snapshot 1`] = `
<div>
  Test
</div>
`;
